src/dashboard/src/components/access/views.py

```diff
@@ -201,6 +212,97 @@ def record_children(client, request, system='', record_id=''):
+    These are returned as dictionaries containing the keys `id`, `label`, and `title`.
+
+    POST:
+    Creates a new digital object component associated with `record_id`.
+    The `label` and `title` of the new object, and the `uuid` of a SIP with which the component should be associated, can be specified by including a JSON document in the request body with values in those keys.
+
+    PUT:
+    Updates an existing digital object component, using `title` and `label` values in a JSON document in the request body.
+    The `component_id` to the component to edit must be specified in the request body.
+    """
+    if request.method == 'POST':
+        record = json.load(request)
+        component = ArchivesSpaceDOComponent.objects.create(
+            resourceid=_normalize_record_id(record_id),
+            sip_id=record.get('uuid'),
+            label=record.get('label', ''),
+            title=record.get('title', ''),
+        )
+
+        try:
+            get_digital_object_component_path(record_id, component.id, system=system, create=True)
+        except ValueError as e:
+            component.delete()
+            response = {
+                'success': False,
+                'message': str(e),
+            }
+            return helpers.json_response(response, status_code=400)
+
+        response = {
+            'success': True,
+            'message': 'Digital object component successfully created',
+            'component_id': component.id,
+        }
+        return helpers.json_response(response)
+    elif request.method == 'GET':
+        components = list(ArchivesSpaceDOComponent.objects.filter(resourceid=_normalize_record_id(record_id), started=False).values('id', 'resourceid', 'label', 'title'))
+        for component in components:
+            access_path = get_digital_object_component_path(record_id, component['id'], system=system)
+            component['path'] = access_path
+            component['type'] = 'digital_object'
+        return helpers.json_response(components)
+    elif request.method == 'PUT':
+        record = json.load(request)
+        try:
+            component_id = record['component_id']
+        except KeyError:
+            response = {
+                'success': False,
+                'message': 'No component_id was specified!'
+            }
+            return helpers.json_response(response, status_code=400)
+
+        try:
+            component = ArchivesSpaceDOComponent.objects.get(id=component_id)
+        except ArchivesSpaceDOComponent.DoesNotExist:
+            response = {
+                'success': False,
+                'message': 'No digital object component exists with the specified ID: {}'.format(component_id),
+            }
+            return helpers.json_response(response, status_code=404)
+        else:
+            component.label = record.get('label', '')
+            component.title = record.get('title', '')
```

Comment:
It is, yes. Both fields are optional, omitting them feels like it should mean "blank" to me.